test/plugins/compliance_visitor_data_request_plugin_test.rb

```diff
@@ -0,0 +1,159 @@
+require "test_helper"
+
+class ComplianceVisitorDataRequestPluginTest < ActiveSupport::TestCase
+    setup do
+        @data_request_plugin = external_api_clients(:compliance_visitor_data_request_plugin)
+        @visitor_api_namespace = api_namespaces(:compliance_visitor_data_request)
+        @visitor_one = api_resources(:compliance_visitor_data_request_one)
+        @visitor_two = api_resources(:compliance_visitor_data_request_two)
+
+        @namespace_with_email_one = api_namespaces(:namespace_with_email_one)
+        @namespace_with_email_two = api_namespaces(:namespace_with_email_two)
+
+        Sidekiq::Testing.fake!
+
+        # How test fixtures are set up:
+        # By default, "compliance_message_sent" of each visitor api resource is false
+        # By default, each visitor has no submission in an api namespace
+    end
+
+    test "Should not send email to visitors if they already received one" do
+        @visitor_one.properties["compliance_message_sent"] = true
+        @visitor_one.save
+        @visitor_two.properties["compliance_message_sent"] = true
+        @visitor_two.save
+
+        assert_no_difference "ActionMailer::Base.deliveries.size" do
+            perform_enqueued_jobs do
+                @data_request_plugin.run
+                Sidekiq::Worker.drain_all
+            end
+        end
+    end
+
+    test "Should send email if visitor did not receive one and has made a submission" do
+        # Visitor one and two are making a submission under namespace_with_email_one
+        ApiResource.create(api_namespace_id: @namespace_with_email_one.id, properties: {"post": "Hello world", "email": @visitor_one.properties["email"]})
+        ApiResource.create(api_namespace_id: @namespace_with_email_one.id, properties: {"post": "Hello world", "email": @visitor_two.properties["email"]})
+        assert_difference "ActionMailer::Base.deliveries.size", +2 do
+            perform_enqueued_jobs do
+                @data_request_plugin.run
+                Sidekiq::Worker.drain_all
+            end 
+        end
+    end
+
+    test "Sender and recipient email addresses and body message are correct" do
+        visitor_email = @visitor_one.properties["email"]
+        ApiResource.create(api_namespace_id: @namespace_with_email_one.id, properties: {"post": "Hello world", "email": visitor_email})
+
+        expected_message = @data_request_plugin.metadata["MESSAGE"]
+
+        perform_enqueued_jobs do
+            @data_request_plugin.run
+            Sidekiq::Worker.drain_all
+        end
+
+        sent_email = ActionMailer::Base.deliveries.last
+        assert(sent_email.from.include?(Subdomain.current.name))
```

Comment:
It looks like this assertion is failing. For some reason, `Subdomain.current.name` is not being included. The test case passed when I ran it locally.

Reply:
@Ayon95 add a `p Subdomain.current.name` and  `p sent_email.from` statements on this line and read what's being printed in the CI log

Reply:
`sent_email.from` is returning an array containing a single email address.

https://github.com/restarone/violet_rails/runs/8239212970?check_suite_focus=true#step:14:70

```
...................................................................................."public"
["public@restarone.com"]
F

Failure:
ComplianceVisitorDataRequestPluginTest#test_Sender_and_recipient_email_addresses_and_body_message_are_correct [/home/runner/work/violet_rails/violet_rails/test/plugins/compliance_visitor_data_request_plugin_test.rb:60]:
Expected false to be truthy.
```

Reply:
@Ayon95 were you able to resolve this? Im not sure why its returning 2 different values, but a hack could be to array wrap it 

 `Array.wrap(sent_email.from).include?(Subdomain.current.name)`

Reply:
Yeah I was also thinking of using `Array.wrap`. Wanted to hear your thoughts on how to handle this.